src/utils/cosmosdb_model.ts

```diff
@@ -79,7 +81,9 @@ export const CosmosResource = t.intersection([
   })
   // this cast is used to keep CosmosResource aligned
   // with @azure/cosmos/Resource type definition
-]) as t.Type<Resource & { readonly id: NonEmptyString }>;
+]) as t.Type<
+  Resource & { readonly id: NonEmptyString; readonly ttl?: NonNegativeNumber }
```

Comment:
Building a readonly BaseModel could make easier the future extension of this typing.
```
const ReadonlyBaseModel = t.readonly(BaseModel);
type ReadonlyBaseModel = t.TypeOf<typeof ReadonlyBaseModel>;
```
```suggestion
  Resource & ReadonlyBaseModel
```

Reply:
I agree with @BurnedMarshal